libsolidity/formal/SMTChecker.cpp

```diff
@@ -114,7 +120,55 @@ bool SMTChecker::visit(FunctionDefinition const& _function)
 		m_loopExecutionHappened = false;
 		m_arrayAssignmentHappened = false;
 	}
+	_function.parameterList().accept(*this);
+	if (_function.returnParameterList())
+		_function.returnParameterList()->accept(*this);
+	visitFunctionOrModifier();
+	return false;
+}
+
+void SMTChecker::visitFunctionOrModifier()
+{
+	solAssert(!m_functionPath.empty(), "");
+	solAssert(!m_modifierDepthStack.empty(), "");
 
+	++m_modifierDepthStack.back();
+	FunctionDefinition const& function = *m_functionPath.back();
+	Block const* codeBlock = nullptr;
+
+	if (m_modifierDepthStack.back() == int(function.modifiers().size()))
+	{
+		if (function.isImplemented())
+			codeBlock = &function.body();
+	}
+	else
+	{
+		solAssert(m_modifierDepthStack.back() < int(function.modifiers().size()), "");
+		ASTPointer<ModifierInvocation> const& modifierInvocation = function.modifiers()[m_modifierDepthStack.back()];
+		solAssert(modifierInvocation, "");
+		modifierInvocation->accept(*this);
+		auto const& modifierDef = dynamic_cast<ModifierDefinition const&>(
+			*modifierInvocation->name()->annotation().referencedDeclaration
+		);
+		std::vector<ASTPointer<Expression>> const& modifierArguments =
+			modifierInvocation->arguments() ? *modifierInvocation->arguments() : std::vector<ASTPointer<Expression>>();
```

Comment:
const ref to a temporary.

Reply:
Which is fine :-).

Reply:
We had that discussion before, hadn't we? Constant references to temporaries will keep the temporary alive for the scope.